addons/web/static/src/views/fields/parsers.js

```diff
@@ -28,28 +28,73 @@ function evaluateMathematicalExpression(expr, context = {}) {
     return evaluateExpr(safeEvalString, context);
 }
 
+/**
```

Comment:
I did not have a full look into your PR, but I have just a small comment here.

Maybe it would be wise to memoize the calls to functions in this file that would compile a regexp.

Not sure if you would need to change their signature by passing multiple string args instead of an object.

I think that compiling regexes is a bit slow, but not entirely sure about this.